### PR TITLE
LPD-43403 Do not rely on kaleoDefinitionVersionId to retrieve the latest version of each workflow definition, rely on version column instead

### PR DESCRIPTION
**Pull request by MarinhoFeliphe**
Related: https://liferay.atlassian.net/browse/LPD-43403

## Situation

The query to retrieve the latest versions of each workflow definition was relying on the `kaleoDefinitionVersionId` instead of the `version`, it will not work well when an upgrade process inserts those definition version asynchronous.

## Fix

Get the latest versions of each workflow definition relying on version column.

### CODE REVIEW

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

**Comment by MarinhoFeliphe** [comment]
ci:test:workflow

File: modules/apps/portal-workflow/portal-workflow-kaleo-service/src/main/java/com/liferay/portal/workflow/kaleo/service/impl/KaleoDefinitionVersionLocalServiceImpl.java [hunk 1]
```diff
@@ -360,62 +361,76 @@ public int getLatestKaleoDefinitionVersionsCount(
...
+						DSLFunctionFactoryUtil.max(
+							DSLFunctionFactoryUtil.castLong(
+								KaleoDefinitionVersionTable.INSTANCE.version)
+						).as(
+							"latestVersion"
+						)
+					).from(
+						KaleoDefinitionVersionTable.INSTANCE
+					).where(
+						KaleoDefinitionVersionTable.INSTANCE.name.eq(
```
Comment (linolaoj) [inline modules/apps/portal-workflow/portal-workflow-kaleo-service/src/main/java/com/liferay/portal/workflow/kaleo/service/impl/KaleoDefinitionVersionLocalServiceImpl.java:421]: Hi @MarinhoFeliphe , should we consider the `companyId` on this inner query?

**Review by linolaoj — CHANGES_REQUESTED** [review]

File: modules/apps/portal-workflow/portal-workflow-kaleo-service/src/main/java/com/liferay/portal/workflow/kaleo/service/impl/KaleoDefinitionVersionLocalServiceImpl.java (hunk 1 above)
Comment (MarinhoFeliphe) [reply modules/apps/portal-workflow/portal-workflow-kaleo-service/src/main/java/com/liferay/portal/workflow/kaleo/service/impl/KaleoDefinitionVersionLocalServiceImpl.java:421]: Yes! good catch!

**Comment by MarinhoFeliphe** [comment]
Resent here: https://github.com/liferay-bpm/liferay-portal/pull/3941

**Pull request closed without merging**